### PR TITLE
Add SonarQube Mapper and OWASP NIST mappings

### PR DESCRIPTION
**Pull request by mirskiy**

### CODE REVIEW

**Comment by mirskiy** [comment]
Added 2 more commits with documentation and updated mappings. Will follow up with a separate issue detailing any additional needed CWE mappings.

**Comment by rx294** [comment]
hey @mirskiy great PR! excellent docs...
just 2 notes
- Rule Severity not populated on to impact field ( hardcoded 0.5 )
rule severity is available on the rule query as below 
```
   "severity"=>"CRITICAL",
```

Severity Levels:
```
Blocker	Operational/security risk: This issue might make the whole application unstable in production. Ex: calling garbage collector, not closing a socket, etc.
Critical	Operational/security risk: This issue might lead to an unexpected behavior in production without impacting the integrity of the whole application. Ex: NullPointerException, badly caught exceptions, lack of unit tests, etc.
Major	This issue might have a substantial impact on productivity. Ex: too complex methods, package cycles, etc.
Minor	This issue might have a potential and minor impact on productivity. Ex: naming conventions, Finalizer does nothing but call superclass finalizer, etc.
Info	Unknown or not yet well defined security risk or impact on productivity.
```
Suggested 'severity' to 'impact' Mapping:

Critical => 'critical'
Blocker  = > 'high'
Major = > 'medium'
Minor = > 'low'
Info = > 'none'


* Rev_* element not included in nest tag...
Heimdal parses expects the last element to be Rev_*

expected nist format:
```
            "tags": {
                "nist": [
                    "SI-10",
                    "AC-12",
                    "Rev_4"
                ]
            },
```

Recommended solution : 
Include a Rev column on the cwe-nist-mapping and owasp-nist-mapping
```
CWE-ID,CWE Name,NIST-ID,Rev,NIST Name
14, Compiler Removal of Code to Clear Buffers,SI-16,4,Memory Protection
```
 and have the mapper include the Rev version as Rev_*

File: lib/heimdall_tools/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module HeimdallTools
-  VERSION = '1.1.1'.freeze
+  VERSION = '2.0.0'.freeze
```
Comment (aaronlippold) [inline lib/heimdall_tools/version.rb]: I think we can just bump to 1.2.0 on this. 2.x would mean a full code rewrite.

**Comment by rx294** [comment]
Approving this PR and the issues will be addresses on a different PR.

**Pull request merged**